File: benchmarks/python/bench_utils.py
import sqlite3
import os
import sys

def get_ext_path():
    if os.name == 'nt':
        return os.path.abspath("../../build/plsqlite.dll")
    elif sys.platform == 'darwin':
        return os.path.abspath("../../build/plsqlite.dylib")
    return os.path.abspath("../../build/plsqlite.so")

def get_db_path():
    use_mem = os.getenv("BENCH_USE_MEMORY", "0") in ("1", "true", "TRUE")
    if use_mem:
        return ":memory:"
    return os.path.abspath("../../databases/bench.db")

def get_connection():
    db_path = get_db_path()
    conn = sqlite3.connect(db_path)

    # If using memory, we need to seed it
    if db_path == ":memory:":
        seed_path = os.path.abspath("../../benchmarks/sql/seed.sql")
        with open(seed_path, 'r') as f:
            conn.executescript(f.read())

    return conn

def load_plsqlite(conn):
    conn.enable_load_extension(True)
    conn.load_extension(get_ext_path())

File: benchmarks/node/bench_utils.js
const Database = require('better-sqlite3');
const path = require('path');
const fs = require('fs');

function getExtPath() {
    const buildDir = path.resolve(__dirname, '../../build');
    if (process.platform === 'win32') return path.join(buildDir, 'plsqlite.dll');
    if (process.platform === 'darwin') return path.join(buildDir, 'plsqlite.dylib');
    return path.join(buildDir, 'plsqlite.so');
}

function getDbConnection() {
    const useMem = process.env.BENCH_USE_MEMORY === '1' || process.env.BENCH_USE_MEMORY === 'true';
    const dbPath = useMem ? ':memory:' : path.resolve(__dirname, '../../databases/bench.db');

    const db = new Database(dbPath);

    if (useMem) {
        const seedPath = path.resolve(__dirname, '../../benchmarks/sql/seed.sql');
        const seedSql = fs.readFileSync(seedPath, 'utf8');
        db.exec(seedSql);
    }

    return db;
}

function loadPLSQLite(db) {
    db.loadExtension(getExtPath());
}

module.exports = {
    getDbConnection,
    loadPLSQLite
};
